src/databricks/labs/lsql/dashboards.py

```diff
@@ -205,131 +736,86 @@ def create_dashboard(self, dashboard_folder: Path) -> Dashboard:
-        except yaml.YAMLError as e:
-            logger.warning(f"Parsing {dashboard_metadata_path}: {e}")
-            return fallback_metadata
-        try:
-            return DashboardMetadata.from_dict(raw)
-        except KeyError as e:
-            logger.warning(f"Parsing {dashboard_metadata_path}: {e}")
-            return fallback_metadata
-
-    def _get_widget(self, widget_metadata: WidgetMetadata) -> Widget:
-        if widget_metadata.is_markdown():
-            return self._get_text_widget(widget_metadata)
-        return self._get_counter_widget(widget_metadata)
-
-    @staticmethod
-    def _get_text_widget(widget_metadata: WidgetMetadata) -> Widget:
-        widget = Widget(name=widget_metadata.id, textbox_spec=widget_metadata.path.read_text())
-        return widget
-
-    def _get_counter_widget(self, widget_metadata: WidgetMetadata) -> Widget:
-        fields = self._get_fields(widget_metadata.path.read_text())
-        query = Query(dataset_name=widget_metadata.id, fields=fields, disaggregated=True)
-        # As far as testing went, a NamedQuery should always have "main_query" as name
-        named_query = NamedQuery(name="main_query", query=query)
-        # Counters are expected to have one field
-        counter_field_encoding = CounterFieldEncoding(field_name=fields[0].name, display_name=fields[0].name)
-        counter_spec = CounterSpec(CounterEncodingMap(value=counter_field_encoding))
-        widget = Widget(name=widget_metadata.id, queries=[named_query], spec=counter_spec)
-        return widget
+        tiles_metadata_with_order = []
+        for order, tile_metadata in enumerate(sorted(tiles_metadata, key=lambda wm: wm.id)):
+            replica = copy.deepcopy(tile_metadata)
+            replica.order = tile_metadata.order if tile_metadata.order is not None else order
+            tiles_metadata_with_order.append(replica)
+
+        tiles, position = [], Position(0, 0, 0, 0)  # Position of first tile
+        for tile_metadata in sorted(tiles_metadata_with_order, key=lambda wm: (wm.order, wm.id)):
+            tile = Tile.from_tile_metadata(tile_metadata)
+            if isinstance(tile, QueryTile):
+                tile.query_transformer = query_transformer
+            placed_tile = tile.place_after(position)
+            tiles.append(placed_tile)
+            position = placed_tile.position
+        return tiles
 
     @staticmethod
-    def _get_fields(query: str) -> list[Field]:
-        parsed_query = sqlglot.parse_one(query, dialect=sqlglot.dialects.Databricks)
-        fields = []
-        for projection in parsed_query.find_all(sqlglot.exp.Select):
-            if projection.depth > 0:
-                continue
-            for named_select in projection.named_selects:
-                field = Field(name=named_select, expression=f"`{named_select}`")
-                fields.append(field)
-        return fields
+    def _get_datasets(tiles: list[Tile]) -> list[Dataset]:
```

Comment:
this method belongs to `DashboardMetadata`

Reply:
See #208